### PR TITLE
Reject overflowing output token estimates before admission

### PR DESCRIPTION
**Pull request by Gajesh2007**
A large positive `n` multiplied by the requested output limit can overflow the generic inference estimate and bypass quota admission. A full or unlimited bucket also admits a saturated maximum-integer estimate, which can then wrap downstream cost and capacity calculations. Reject the invalid product with 400 `invalid_request_error` and `param: "n"` before quota, balance reservation or routing.

The shared estimator selects the existing explicit/default bound once and reports whether multiplication fits. Both inference callers use the same validation response. Fitting products retain their existing estimate, forwarded generation limits and one-burst quota policy. This PR handles multiplication overflow; #943 separately hardens prompt-plus-output sums. Existing numeric conversions and independently huge representable limits are outside this change.

```mermaid
flowchart LR
  subgraph Before
    A[Generic completions or messages] --> B[estimateRequestedMaxTokens]
    B --> C[Unchecked n times token bound]
    C --> D[Wrapped estimate reaches quota and billing]
  end
  subgraph After
    E[Inference request] --> F[estimateRequestedMaxTokens reports value and validity]
    F --> G[validateRequestedMaxTokens]
    G -->|overflow| H[400 invalid_request_error, param n]
    G -->|fits| I[Existing quota, reservation and routing]
  end
```

Validation:

- Real HTTP fixtures reproduce positive and negative wraps on both generic routes against master. Existing half-burst and ordinary oversized controls remain.
- Twelve additional HTTP cases reproduce the full/unlimited-quota issue against the earlier PR head (0.702s): both generic routes, both quota states, positive/negative/default-bound overflow. Every case reached billing and recorded a minimum-charge reserve plus refund. The final fixture requires 400 and only the original deposit in the real MemoryStore ledger.
- Boundary cases preserve normal products, the largest fitting product, all three limit field names and the default bound. Final focused API race tests pass in 0.700s. The final full API race suite passes in 228.309s with database environment variables unset.
- Two independent source/test reviews are clean. Signed commits, `git diff --check` and 278-page docs lint pass. API, consumer and routing docs describe the rejection boundary.

### CODE REVIEW

File: coordinator/api/request_introspection.go [hunk 1]
```diff
@@ -540,18 +541,22 @@ func requestHasTools(parsed map[string]any) bool {
...
+			maxTokens = n
+			break
 		}
 	}
 	if copies, ok := intFromRequestValue(parsed["n"]); ok && copies > 1 {
-		return 256 * copies
+		// An overflowing estimate must stay conservative for token quotas,
+		// reservations and routing, never wrap into a smaller request.
+		if maxTokens > math.MaxInt/copies {
+			return math.MaxInt
```
[the remark this answers is not in the document]
Comment (Gajesh2007) [reply coordinator/api/request_introspection.go]: Fixed in 932037b7c (final head aa3f30c7d). The estimator now reports invalid multiplication explicitly, and both API callers return 400 invalid_request_error with param n before token admission, balance reservation or routing. No saturated token value reaches downstream arithmetic. Twelve real HTTP cases reproduced the issue on the prior head with full/unlimited quotas and actual minimum-charge reserve/refund ledger entries; the final fixture requires 400 and no added ledger records. Fitting products keep their existing behavior. Focused race tests, the full API race suite (228.309s), two independent reviews and docs lint pass.

**Comment by Gajesh2007** [comment]
@codex review

**Comment by Gajesh2007** [comment]
Consolidated into #1056 for review and merge after #1094, #1095 and #1096. #1056 contains this branch's exact head (`aa3f30c7d726bc8311178a852b4a94dd6f5a437b`), including its reviewed fixes, with implementations and regressions reconciled into the coordinator subsystem owners. Closing the component PR to collapse the stack; its branch, commits and review history are retained. Follow #1056 for combined validation and merge status.
